app/lib/actions/katello/host/update.rb

```diff
@@ -12,11 +12,11 @@ def plan(host, consumer_params = nil)
             host.content_facet.save! if host.content_facet
 
             if host.subscription_facet
-              if consumer_params
-                host.subscription_facet.update_from_consumer_attributes(consumer_params)
-              else
+              unless consumer_params
                 consumer_params = host.subscription_facet.consumer_attributes
               end
+
+              host.subscription_facet.update_from_consumer_attributes(consumer_params)
```

Comment:
This was based on feedback from @thomasmckay here (https://github.com/ehelms/katello/pull/16/files/c4e54e9faef0fe6de572efba3aa99431fdbd7f64#diff-0814801452c6a2779e2bf22963f6bd58) related to his testing trying to do updates via host API.